### ISSUE TITLE
Corner cases that fail for lack of `LD_LIBRARY_PATH`

### ISSUE THREAD

**Issue by ocaisa** [EESSI/software-layer#192]
Python packages may use ctypes to load libraries:
```
cdll.LoadLibrary("libprotoc.so.3.13.0.0")
```
but this relies on the linker being able to find the libraries, which can fail in our current setup.

**Comment by mboisson** [comment]
This will be relevant to you: https://github.com/ComputeCanada/software-stack/issues/100

**Comment by mboisson** [comment]
Relevant issue in cpython upstream: https://github.com/python/cpython/issues/65241

**Comment by ocaisa** [comment]
@mboisson I'm thinking about making a trivial update to `ctypes` via a sed command that would change https://github.com/python/cpython/blob/main/Lib/ctypes/util.py#L316 to
```
libpath = os.environ.get('LD_LIBRARY_PATH', os.environ.get('LIBRARY_PATH'))
```
that should also solve the issue in our case I think.

**Comment by boegel** [comment]
@ocaisa Why don't we install that as a proper patch for the Python installations we have in EESSI (or even always, I guess it wouldn't do any harm)?

**Comment by ocaisa** [comment]
Yes sure, I was thinking of doing it via `sed ... &&` in `preconfigopts`

**Comment by ocaisa** [comment]
I guess I could actually go ahead and change this in the easyblock. The only scenario I can see this being a problem is where you break the detection by defining `LD_LIBRARY_PATH`

**Comment by ocaisa** [comment]
See https://github.com/easybuilders/easybuild-easyblocks/pull/3352

**Comment by mboisson** [comment]
> @mboisson I'm thinking about making a trivial update to `ctypes` via a sed command that would change https://github.com/python/cpython/blob/main/Lib/ctypes/util.py#L316 to
> 
> ```
> libpath = os.environ.get('LD_LIBRARY_PATH', os.environ.get('LIBRARY_PATH'))
> ```
> 
> that should also solve the issue in our case I think.

that would not fix anything if both LD_LIBRARY_PATH and LIBRARY_PATH are defined.

**Comment by mboisson** [comment]
In my opinion, the correct fix is that `find_library` needs to return the full path. It actually does find the library through `_findLib_gcc`, since it calls the compilers which relies on `LIBRARY_PATH`, but it then returns only the file name of the SO instead of the full path as it should.

**Comment by ocaisa** [comment]
PR is at https://github.com/easybuilders/easybuild-easyblocks/pull/3352 , I've only triggered this in specific circumstances and respect the use of both `LD_LIBRARY_PATH` and `LIBRARY_PATH` in those circumstances

**Comment by casparvl** [comment]
See also https://gitlab.com/eessi/support/-/issues/154
